components/function-controller/internal/controllers/serverless/reconcile.go

```diff
@@ -0,0 +1,141 @@
+package serverless
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/kyma/components/function-controller/internal/resource"
+	serverlessv1alpha1 "github.com/kyma-project/kyma/components/function-controller/pkg/apis/serverless/v1alpha1"
+)
+
+const (
+	functionNameLabel      = "serverless.kyma-project.io/function-name"
+	functionManagedByLabel = "serverless.kyma-project.io/managed-by"
+	functionUUIDLabel      = "serverless.kyma-project.io/uuid"
+
+	serviceBindingUsagesAnnotation = "servicebindingusages.servicecatalog.kyma-project.io/tracing-information"
+
+	cfgGenerationLabel = "serving.knative.dev/configurationGeneration"
+
+	configMapFunction = "handler.js"
+	configMapHandler  = "handler.main"
+	configMapDeps     = "package.json"
+)
+
+var (
+	envVarsForRevision = []corev1.EnvVar{
+		{Name: "FUNC_HANDLER", Value: "main"},
+		{Name: "MOD_NAME", Value: "handler"},
+		{Name: "FUNC_TIMEOUT", Value: "180"},
+		{Name: "FUNC_RUNTIME", Value: "nodejs12"},
+		// {Name: "FUNC_MEMORY_LIMIT", Value: "128Mi"},
+		{Name: "FUNC_PORT", Value: "8080"},
+		{Name: "NODE_PATH", Value: "$(KUBELESS_INSTALL_VOLUME)/node_modules"},
+	}
+)
+
+type FunctionReconciler struct {
+	client.Client
+	Log logr.Logger
+
+	resourceClient resource.Resource
+	recorder       record.EventRecorder
+	config         FunctionConfig
+	scheme         *runtime.Scheme
+}
+
+func NewFunction(client client.Client, log logr.Logger, config FunctionConfig, scheme *runtime.Scheme, recorder record.EventRecorder) *FunctionReconciler {
+	resourceClient := resource.New(client, scheme)
+
+	return &FunctionReconciler{
+		Client:         client,
+		Log:            log,
+		config:         config,
+		resourceClient: resourceClient,
+		scheme:         scheme,
+		recorder:       recorder,
+	}
+}
+
+func (r *FunctionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&serverlessv1alpha1.Function{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&batchv1.Job{}).
+		Owns(&servingv1.Service{}).
+		Owns(&servingv1.Revision{}).
```

Comment:
this line is the only one I added while moving this functionality to separate file